presto-main/src/main/java/com/facebook/presto/metadata/MetadataManager.java

```diff
@@ -1216,6 +1217,16 @@ private boolean canResolveOperator(OperatorType operatorType, List<TypeSignature
         }
     }
 
+    private ConnectorTableLayoutHandle getTableLayout(Session session, TableHandle tableHandle)
+    {
+        if (tableHandle.getLayout().isPresent()) {
+            return tableHandle.getLayout().get();
+        }
+        Optional<TableLayoutResult> result = getLayout(session, tableHandle, alwaysTrue(), Optional.empty());
```

Comment:
The function names here start to get really confusing.
* There are two `getLayout` now but they return different types. Maybe rename this one `getLayoutResult`.
* THere are two `getTableLayout` and the `ConnectorMetadata.getTableLayout` returns a `ConnectorTableLayout` while this function returns `ConnectorTableLayoutHandle`. Maybe just name this function `getConnectorTableLayoutHandle`.

Reply:
Yes, TableLayout will be renamed to TableProperties later. ConnectorTableLayout will become ConnectorTableProperties.